src/SIM/ModelGenerator.h

```diff
@@ -21,127 +21,112 @@
 class SIMbase;
 class TiXmlElement;
 
+
 /*!
   \brief Base class for model generators for FEM simulators.
 */
 
 class ModelGenerator
 {
 public:
-  //! \brief Constructor initializes common members
+  //! \brief The constructor initializes the common members.
   //!\ param elem XML element to parse
-  ModelGenerator(const TiXmlElement* elem);
-
+  ModelGenerator(const TiXmlElement* elem) : geo(elem) {}
   //! \brief Empty destructor.
   virtual ~ModelGenerator() {}
 
   //! \brief Creates a geometry.
-  //! \param[in] sim SIM with patch read function to use
-  virtual SIMdependency::PatchVec createGeometry(const SIMbase& sim) const = 0;
+  //! \param[in] m Simulator object with patch read function to use
+  SIMdependency::PatchVec createGeometry(const SIMbase& m) const;
 
-  //! \brief Creates topology for geometry.
-  //! \param[in] geo XML element containing geometry defintion
-  //! \param sim Simulator to apply topology to
-  virtual bool createTopology(SIMbase& sim) const = 0;
+  //! \brief Creates topology for multi-patch geometries.
+  virtual bool createTopology(SIMbase&) const { return true; }
 
   //! \brief Creates topology sets for geometry.
-  //! \param[in] sim Simulator with patch ownerships
+  //! \param[in] sim Simulator object with patch ownerships
   virtual TopologySet createTopologySets(const SIMbase& sim) const = 0;
 
 protected:
-  bool sets; //!< Whether to generate topologysets or not
+  //! \brief Generates the G2 description of the geometry.
+  //! \param nsd Number of spatial dimension
+  virtual std::string createG2(int nsd) const = 0;
```

Comment:
OK, så du sier det vil bli andre ModelGenerator sub-klasser som ikke bruker createG2. Den kunne jeg ikke se komme, og mitt forslag tok ikke høyde for dette. Men det er vel greit å ha en tom default implementasjon i stedet. Se fixup commit.